### ISSUE TITLE
globs with multiple expanded parts broken in 3.2.0

### ISSUE THREAD

**Issue by devongovett** [mrmlnc/fast-glob#257]
### Environment

  * OS Version: macOS 10.12
  * Node.js Version: 12

## Actual Behavior

Patterns with multiple glob parts that are broken now.

fast-glob@3.2.0:

```js
require('fast-glob').sync('packages/@react-spectrum/*/docs/*.mdx')
[]
```

## Expected Behavior

fast-glob@3.1.1:

```js
require('fast-glob').sync('packages/@react-spectrum/*/docs/*.mdx')
[
  '/Users/govett/dev/react-spectrum/packages/@react-spectrum/button/docs/Button.mdx',
  '/Users/govett/dev/react-spectrum/packages/@react-spectrum/radio/docs/Radio.mdx',
  '/Users/govett/dev/react-spectrum/packages/@react-spectrum/textfield/docs/TextArea.mdx',
  '/Users/govett/dev/react-spectrum/packages/@react-spectrum/textfield/docs/TextField.mdx'
]
```

**Comment by mrmlnc** [comment]
Just checked the latest version on Windows 10 and WSL (Ubuntu).

Unfortunately, I cannot reproduce your problem. Any additional steps? Maybe you have a repository?

```
packages/
└── @react-spectrum
    ├── button
    │   └── docs
    │       └── test.mdx
    └── radio
        └── docs
            └── test.mdx
```

```js
const fg = require('.');

const entries = fg.sync('packages/@react-spectrum/*/docs/*.mdx');

console.dir(entries, { colors: true });
```

```js
[
  'packages/@react-spectrum/button/docs/test.mdx',
  'packages/@react-spectrum/radio/docs/test.mdx'
]
```

**Comment by milesj** [comment]
I may be having a similar problem? Updated to 3.2.1 and this now fails and returns no results:

```js
fg.sync([`./packages/*/package.json`], {
  absolute: true,
  cwd: '~/repos/nimbus', // is absolute in code
});
```

**Comment by mrmlnc** [comment]
Thanks, @milesj! I'll check it today.

**Comment by mrmlnc** [comment]
Confirm. The pattern with leading dot works incorrectly for deep reading. I'll fix it today.

**Comment by mrmlnc** [comment]
I'm published a release ([`3.2.2`](https://github.com/mrmlnc/fast-glob/releases/tag/3.2.2)) addressed to this issue. If you have any problems, please create a separated issue.

Thanks for the feedback :tada:

**Comment by ondratra** [comment]
Is this issue really fixed in `3.2.2`? I have upgraded from some older version of `fast-glob` to `3.2.2` but group expansion is broken (`somaPath/(option1|option2)/*.extension` finds nothing even when `somaPath/option1/*.extension` finds some files)
